scrapy/utils/versions.py

```diff
@@ -1,31 +1,48 @@
+from __future__ import annotations
+
 import platform
 import sys
+from importlib.metadata import version
+from warnings import warn
 
-import cryptography
-import cssselect
 import lxml.etree
-import parsel
-import twisted
-import w3lib
 
-import scrapy
+from scrapy.exceptions import ScrapyDeprecationWarning
+from scrapy.settings.default_settings import LOG_VERSIONS
 from scrapy.utils.ssl import get_openssl_version
 
+_DEFAULT_SOFTWARE = ["Scrapy"] + LOG_VERSIONS
+
+
+def _version(component):
+    lowercase_component = component.lower()
+    if lowercase_component == "libxml2":
+        return ".".join(map(str, lxml.etree.LIBXML_VERSION))
+    if lowercase_component == "platform":
+        return platform.platform()
+    if lowercase_component == "pyopenssl":
+        return get_openssl_version()
+    if lowercase_component == "python":
+        return sys.version.replace("\n", "- ")
+    return version(component)
+
+
+def get_versions(
+    software: list | None = None,
+) -> list[tuple[str, str]]:
+    software = software or _DEFAULT_SOFTWARE
+    return [(item, _version(item)) for item in software]
+
 
-def scrapy_components_versions() -> list[tuple[str, str]]:
-    lxml_version = ".".join(map(str, lxml.etree.LXML_VERSION))
-    libxml2_version = ".".join(map(str, lxml.etree.LIBXML_VERSION))
-
-    return [
-        ("Scrapy", scrapy.__version__),
-        ("lxml", lxml_version),
-        ("libxml2", libxml2_version),
-        ("cssselect", cssselect.__version__),
-        ("parsel", parsel.__version__),
-        ("w3lib", w3lib.__version__),
-        ("Twisted", twisted.version.short()),
-        ("Python", sys.version.replace("\n", "- ")),
-        ("pyOpenSSL", get_openssl_version()),
-        ("cryptography", cryptography.__version__),
-        ("Platform", platform.platform()),
-    ]
+def scrapy_components_versions(
+    components: list | None = None,
```

Comment:
What do you think about not adding this argument to scrapy_components_version? In this case, implementation will be different: it'd call `get_versions(_DEFAULT_SOFTWARE)`. We still should deprecate it probably.